src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -60,16 +61,19 @@ pub struct CloudwatchLogsSinkConfig {
     pub create_missing_stream: Option<bool>,
     #[serde(default, flatten)]
     pub batch: BatchConfig,
-
-    // Tower Request based configuration
-    pub request_in_flight_limit: Option<usize>,
-    pub request_timeout_secs: Option<u64>,
-    pub request_rate_limit_duration_secs: Option<u64>,
-    pub request_rate_limit_num: Option<u64>,
-    pub request_retry_attempts: Option<usize>,
-    pub request_retry_backoff_secs: Option<u64>,
+    #[serde(default, flatten)]
```

Comment:
Just FYI, I've seen serde do some slightly unexpected things when combining `default` and `flatten`. It could be a good idea to have a test or two that ensure you can pass none, some, or all of the options and get the expected behavior.

Reply:
Are you thinking of a test for the `TowerRequestConfig` structure itself, or tests within sinks to ensure proper config parsing?

Reply:
Can't `default` annotation be safely removed from here, keeping the code behavior intact? If it is removed, all fields of `request` should be set to their default values if not specified, which are `None` for `Option`.

Reply:
Yes, apparently this is superfluous with `flatten` (tested with a simple test harness as well as full Vector).